lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -38,100 +38,39 @@
-                master_coord, coordinates))
-        raise ValueError(msg)
-
-    # If the master_coord is not a dimension coordinate, then the other
-    # coordinates cannot be associated with it.
-    if len(cube.coords(master_coord, dim_coords=True)) > 0:
-        for coord in coordinates:
-            if cube.coords(coord):
-                temp_coord = cube.coord(coord)
-                cube.remove_coord(coord)
-                temp_aux_coord = (
-                    build_coordinate(temp_coord.points,
-                                     bounds=temp_coord.bounds,
-                                     coord_type=AuxCoord,
-                                     template_coord=temp_coord))
-                coord_names = [
-                    coord.standard_name for coord in cube.dim_coords]
-                cube.add_aux_coord(
-                    temp_aux_coord,
-                    data_dims=coord_names.index(master_coord))
-    return cube
-
-
-def _slice_over_coordinate(cubes, coord_to_slice_over):
-    """
-    Function slice over the requested coordinate,
-    promote the sliced coordinate into a dimension coordinate
-    to help concatenation.
+    Function to remove attributes that do not match between all cubes on the
+    list.  Cubes are modified in place.
 
     Args:
-        cubes (Iris cubelist or Iris cube):
-            Cubes to be concatenated.
-        coords_to_slice_over (List):
-            Coordinates to be sliced over.
+        cubes (iris.cube.CubeList):
+            List of cubes to check the attributes and revise.
 
-    Returns:
-        Iris CubeList
-            CubeList containing sliced cubes.
+    Kwargs:
+        silent (list or None):
+            List of attributes to remove silently if unmatched.
 
+    Warns:
+        UserWarning:
+            If an unmatched attribute is not on the "silent" list,
+            a warning will be raised.
     """
-    sliced_by_coord_cubelist = iris.cube.CubeList([])
-    if isinstance(cubes, iris.cube.Cube):
-        cubes = iris.cube.CubeList([cubes])
-
-    for cube in cubes:
-        if cube.coords(coord_to_slice_over):
-            for coord_slice in cube.slices_over(coord_to_slice_over):
-                coord_slice = iris.util.new_axis(
-                    coord_slice, coord_to_slice_over)
-                sliced_by_coord_cubelist.append(coord_slice)
-        else:
-            sliced_by_coord_cubelist.append(cube)
-
-    return sliced_by_coord_cubelist
+    unmatched = compare_attributes(cubes)
+    warning_msg = 'Deleting unmatched attribute {}, value {}'
+    if len(unmatched) > 0:
+        for i, cube in enumerate(cubes):
+            for attr in unmatched[i]:
+                if silent is not None and attr in silent:
+                    cube.attributes.pop(attr)
+                else:
+                    warnings.warn(
+                        warning_msg.format(attr, cube.attributes[attr]))
+                    cube.attributes.pop(attr)
```

Comment:
This can be simplified:
```
if silent is not None and attr not in silent:
    warnings.warn(
        warning_msg.format(attr, cube.attributes[attr]))
    cube.atributes.pop(attr)
```